extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -666,6 +670,30 @@ public String getEarliestSequenceNumber(StreamPartition<String> partition)
     return getSequenceNumber(partition, ShardIteratorType.TRIM_HORIZON);
   }
 
+  @Override
+  public boolean isOffsetAvailable(StreamPartition<String> partition, OrderedSequenceNumber<String> offset)
```

Comment:
This is quite different from the current logic in KinesisRecordSupplier#getSequenceNumber. 
such as handling when `closed` or the Retry for kinesis.getRecords

Reply:
The previous exception handling has been added